yaml_edit/lib/src/editor.dart

```diff
@@ -244,7 +244,7 @@ class YamlEditor {
   ///   - test
   ///   - 2
   /// ```
-  void update(Iterable<Object> path, Object value) {
+  void update(Iterable<Object?> path, Object? value) {
```

Comment:
This is the only interface change for `YamlEditor` that was necessary to conform with the tests on `test/special_test.dart`:

```dart
doc.update([null], 'tilde');
```
and on `test/update_test.dart` on line 839:
```dart
doc.update(['Mark McGwire'], null);
```

Reply:
If we allow `null` in `path` arguments then we should do so for all methods that take a `path`.

Looking at the special test, I guess this is allowed because keys in YAML dictionaries can be `null`.

From the test case:
```dart
    final doc = YamlEditor('''
~: null # <-- The ~ character means null
false: false
No: No
true: true
''');
    doc.update([null], 'tilde');
    doc.update([false], false);
    doc.update(['No'], 'no');
    doc.update([true], 'true');

    expect(doc.toString(), equals('''
~: tilde # <-- The ~ character means null
false: false
No: no
true: "true"
'''));
```

Took a bit of googling to find out that [tilde is null in YAML](https://stackoverflow.com/questions/51990175/what-is-the-purpose-of-tilde-character-in-yaml).

Yes, YAML dictionaries don't just map from string to values, keys can be anything, even `null`. So I guess `null` is allowed in paths.

That probably also applied to `parseAt` and other methods that accept a `path`.

Reply:
As for `doc.update(['Mark McGwire'], null);`, yes obviously, it's possible to set a key in YAML document to `null`. That makes sense.

Again, anywhere a value is inserted or replace we probably have to support inserting `null`. Like `appendToList` and `spliceList`, why shouldn't I be able to insert `null` into a list?

PS. feel free to add more special test cases to cover things we don't have tests for :D